native/src/main/java/io/ballerina/observe/trace/newrelic/NewRelicTracerProvider.java

```diff
@@ -0,0 +1,116 @@
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.observability.tracer.spi.TracerProvider;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+import java.io.PrintStream;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+
+/**
+ * This is the New Relic tracing extension class for {@link TracerProvider}.
+ */
+public class NewRelicTracerProvider implements TracerProvider {
+    private static final String TRACER_NAME = "newrelic";
+    private static final String TRACE_REPORTER_ENDPOINT = "https://otlp.nr-data.net:4317";
+    private static final String TRACE_API_KEY_HEADER = "Api-Key";
+
+    static SdkTracerProviderBuilder tracerProviderBuilder;
+
+    @Override
+    public String getName() {
+        return TRACER_NAME;
+    }
+
+    @Override
+    public void init() {
+        // Do Nothing
+    }
+
+    public static BArray startPublishingTraces(BString apiKey, BString samplerType, BDecimal samplerParam,
+                                                int reporterFlushInterval, int reporterBufferSize) {
+        BArray output = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_STRING));
+
+        OtlpGrpcSpanExporter exporter = OtlpGrpcSpanExporter.builder()
+                .setEndpoint(TRACE_REPORTER_ENDPOINT)
+                .addHeader(TRACE_API_KEY_HEADER, apiKey.getValue())
+                .build();
+
+        tracerProviderBuilder = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor
+                        .builder(exporter)
+                        .setMaxExportBatchSize(reporterBufferSize)
+                        .setExporterTimeout(reporterFlushInterval, TimeUnit.MILLISECONDS)
+                        .build());
+
+        tracerProviderBuilder.setSampler(selectSampler(samplerType, samplerParam));
+        output.append(StringUtils.fromString("ballerina: started publishing traces to New Relic on " +
+                TRACE_REPORTER_ENDPOINT));
+
+        return output;
+    }
+
+    private static Sampler selectSampler(BString samplerType, BDecimal samplerParam) {
+        switch (samplerType.getValue()) {
+            default:
+            case "const":
```

Comment:
Can we move this into a constant like in [Ballerina](https://github.com/ballerina-platform/module-ballerinax-newrelic/pull/1/files#diff-a6eec05028577b363a8e5ddee160ce99e970cbf813360e7743f1277f5836bbd1R22)